### ISSUE TITLE
Request: make the shifting maze bigger

### ISSUE THREAD

**Issue by Builderb0y** [ReikaKalseki/Reika_Mods_Issues#945]
Currently, the passageways of of the shifting maze are only one block wide, and 2 blocks tall. It would be much nicer to walk around in if they were 2x3 or larger.

**Comment by HellFirePvP** [comment]
It'd make finding the chests with the keys easier from above in my opinion, so not sure if that's what you want to do.

Other than that, i could do that and provide a PR if that's in your interest Reika? I'm also sure to do proper testing before sending u the PR.

**Comment by ReikaKalseki** [comment]
...Do you even know how the code works?

**Comment by HellFirePvP** [comment]
Working with java and minecraft for many years now and i've looked into your code quite a bit. Especially since i was extremly confused about all the cc dimension stuff at first and spent some hours looking into the src to find how things are generated, since even with using dimping and flying around wherever it showed me the symbols i couldn't find the entrance to certain structures... - including finding the monument. (The cameraflight for the ritual still doesn't work properly though...)

So yea. I'm fairly certain i can do that, if you allow me to. (Might take 1-2 days tho since i'm not on my main computer atm)

Edit:Looking into it, it does certainly look a bit more challenging to expand, i'd like to take the challenge, but i can see where your concern might come from. However, i can assure you i know what i'm doing.

Edit2:It looks more like you gave up on the dynamic shift-maze generation and did a fixed one instead... So what should i adapt? Should i try make a dynamically generating one or sit down and generate a few thousand lines of code that generate a bigger-sized one...

**Comment by ReikaKalseki** [comment]
If you can find a solution to a dynamically generating one, I am certainly interested to hear it. But as of right now, the fixed one is the one that was implemented.

The problem with the dynamic one was twofold.

Problem one was getting the right amount of "overlap" between phases. Like the fixed maze, I want the different states of the maze to be 90% identical, but with a few key locations opening or closing. All attempts I tried either gave 100% congruence, <20% congruence, or simply froze (i.e. reached infinite loops or so close to it that it was indistinguishable).

Problem two is the 'anchors' as they were called in the code. The original design had 3x3 "rooms" with a pyramid in the center as the key areas. Building "around" those rooms such that they only had one entrance, and where only one was accessible per path, proved insurmountable.

**Comment by HellFirePvP** [comment]
I'll do some thought experiments and see if i can come up with some sort of algorithm to generate some maze like you mentioned. It seems that it looks even harder to do than it sounds like, but i'm certain that there is a "good" way to get a maze like the one you want to have. I'm gonna spend some days building a model around that.

Should i just reply here once i come up with something or would you prefer some other way of communication?

One honest question though: Did you write all that fixed maze by hand? It doesn't look like it but i wonder if you wrote a small program to generate the code for it or if there's something public out there i just can't find... (However, if you wrote it by hand, mad respects. I would've gone crazy)

**Comment by ReikaKalseki** [comment]
> Should i just reply here once i come up with something or would you prefer some other way of communication?

I have a teamspeak which I suppose could work, assuming that voice chat is feasible. Do note that my scheduling there, while normally very regular, is irregular for the next week or so. Today, I will be there in an hour or so, and though I would normally be on for eight hours continuously, there are likely to be frequent and possibly lengthy interruptions.

> One honest question though: Did you write all that fixed maze by hand? It doesn't look like it but i wonder if you wrote a small program to generate the code for it or if there's something public out there i just can't find... (However, if you wrote it by hand, mad respects. I would've gone crazy)

I planned it all out and initially built it by hand, years ago for a puzzle map I made. I then converted it through MCEdit schematic and Mithion's Schematic-To-Java tool, which generates semiusable code (and is also in use for other CC structures, as well as several ThaumCraft ones).
